Detectors/TRD/reconstruction/src/CruRawReader.cxx

```diff
@@ -114,26 +114,45 @@ void CruRawReader::OutputHalfCruRawData()
   LOG(info) << "Full 1/2 CRU dump end ***************";
 }
 
+void CruRawReader::dumpRDHAndNextHeader(const o2::header::RDHAny* rdh)
+{
+  LOG(info) << "######################### Dumping RDH incoming buffer ##########################";
+  o2::raw::RDHUtils::printRDH(rdh);
+  LOG(info) << "Now for the buffer breakdown";
+  auto offsetToNext = o2::raw::RDHUtils::getOffsetToNext(rdh);
+  for (int i = 0; i < offsetToNext / 4; ++i) {
+    LOG(info) << "ii:" << i << " 0x" << std::hex << *((uint32_t*)rdh + i); //*(uint32_t*)(reinterpret_cast<const uint32_t*>(rdh) + i);
+  }
+  LOG(info) << "Next 8....";
+  for (int i = 0; i < 8; ++i) {
+    LOG(info) << "iii:" << i << " 0x" << std::hex << *((uint32_t*)rdh + i + offsetToNext); //*(uint32_t*)(reinterpret_cast<const uint32_t*>(rdh) + i+offsetToNext);
+  }
+  LOG(info) << "######################### Finished RDH incoming buffer ##########################";
+}
+
 bool CruRawReader::processHBFs(int datasizealreadyread, bool verbose)
 {
   if (mHeaderVerbose) {
-    LOG(info) << "PROCESS HBF starting at " << std::hex << (void*)mDataPointer;
+    LOG(info) << "PROCESS HBF starting at " << std::hex << (void*)mDataPointer << " already read in : " << datasizealreadyread;
   }
   mDataRDH = reinterpret_cast<const o2::header::RDHAny*>(mDataPointer);
-  mOpenRDH = reinterpret_cast<o2::header::RDHAny*>((char*)mDataPointer);
+  mOpenRDH = reinterpret_cast<const o2::header::RDHAny*>((const char*)mDataPointer);
```

Comment:
Ok, as I said I don't really understand most of this pointer casting, like here the `reinterpret_cast<>` with an intermediate cast to `char*`. And also the rest here is very hard to understand for me. But if you have tested this and it fixes an actual issue then it is fine with me.

Reply:
;-) that is actually not used, a legacy from testing early on and how I was intending to go through the rdhs comprising the relevant data. I ended up simply combining the payload parts for each rdh into a singular cru payload and working with that.
The change in this pr is actually accidental. New reader is way way more clear as to what is going on, and faster.